docs/admonitions.rst

```diff
@@ -2,219 +2,309 @@
+Admonition Directive
 *********************
 
-.. _sphinxcontrib-details-directive extension: https://pypi.org/project/sphinxcontrib-details-directive
+All admonitions listed above are created or overridden by this theme's `Custom Admonitions`_ feature.
+This means that the following options are available to all admonitions.
 
-For collapsible dropdown admonitions, the mkdocs-material theme relies on a markdown syntax
-extension that cannot be used with sphinx. Instead, this sphinx-immaterial theme relies on
-the `sphinxcontrib-details-directive extension`_
-to get similar results.
+.. rst:directive:: admonition
 
-The `sphinxcontrib-details-directive extension`_ should be added to conf.py's extension list.
+   This directive was overridden from the :dudir:`docutils definition <generic-admonition>`.
+   The only difference (aside from the added options listed below) is that the
+   inherently required argument used for a title has been made optional.
 
-.. code-block:: python
+   .. rst:directive:option:: title
 
-   extensions = ["sphinx_immaterial", "sphinxcontrib.details.directive"]
+      This option provides an alternative for custom titles in admonitions defined in
+      `rST and Sphinx`_ because they don't accept any directive arguments. When used with the
+      generic :rst:dir:`admonition` and the inherited `Admonitions from mkdocs-material`_, the
+      value for this option is concatenated with a title given as a directive argument.
 
-If the :rst:`:class:` option is not supplied to the ``details`` directive then the admonition
-style falls back to a `note` admonition style.
+      The :rst:`:no-title:` option will supersede any given title.
 
-.. rst-example::
+      .. rst-example:: Equivalent ways to customize the admonition's title.
 
-   .. details:: Open by default
-      :class: example
-      :open:
+         .. tip::
+            :title: A custom title specified in the directive's :rst:`:title:` option.
 
-      Use the :rst:`:open:` option as a flag to expand the admonition by default.
+            The :dutree:`tip` directive accepts no arguments.
 
-.. rst-example::
+         .. admonition:: A custom title specified in the directive's *argument*.
 
-   .. details:: Closed by default
-      :class: help
+            The default style for the generic admonition is that of the `note admonition <note-style>`.
 
-      Without the :rst:`:open:` flag, the admonition is collapsed by default.
+         .. example-admonition:: A custom title specified in both the
+            :title: directive's *argument* and :rst:`:title:` option.
+               It can even span multiple lines.
 
-Removing the title
-******************
+            Notice the blank line between the directive's beginning block and this content block.
+
+            This admonition's directive was created just for this documentation using the
+            sphinx-immaterial theme's `Custom Admonitions`_ feature.
+
+   .. rst:directive:option:: no-title
```

Comment:
Perhaps we don't need this option --- instead just specifying an empty title with `:title:` could serve the same purpose.

Also, if the title is specified both as the argument and as this option, perhaps it should just result in an error rather than concatenating the two; that would be more consistent behavior, I think, if we decide to have an empty `:title:` mean no title.

Reply:
I actually had it to do that when I first wrote it, but an empty value to a `title` option doesn't feel user intuitive enough. It didn't seem too radical to have a `no-title` flag because this is similar to what autodoc does for some it's directive's options.

The way it is now felt more natural to write in RST.

Reply:
Okay, makes sense.